apollo-test/src/test/java/com/spotify/apollo/test/helper/MatchersHelper.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * -\-\-
+ * Spotify Apollo Testing Helpers
+ * --
+ * Copyright (C) 2013 - 2017 Spotify AB
+ * --
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * -/-/-
+ */
+package com.spotify.apollo.test.helper;
+
+import org.hamcrest.Matcher;
+import org.hamcrest.StringDescription;
+
+import static org.junit.Assert.assertThat;
+
+public final class MatchersHelper {
+  private MatchersHelper() {
+  }
+
+  public static <T> void assertDoesNotMatch(final T value, final Matcher<T> matcher) {
+    try {
+      assertThat(value, matcher);
+    } catch (final AssertionError ignored) {
+      return;
+    }
+    StringDescription description = new StringDescription();
+    matcher.describeTo(description);
+    throw new AssertionError("Expected " + value + " to not match: " + description.toString());
```

Comment:
Yes, but `not` does not report the output of describeTo properly, which causes problems with coverage. 

Also, this makes it easy to verify the error descriptions work as needed.